compact.go

```diff
@@ -238,6 +247,33 @@ func (c *LeveledCompactor) selectDirs(ds []dirMeta) []dirMeta {
 	return nil
 }
 
+// selectOverlappingDirs returns one set of dir metas of the blocks whose ranges are overlapping,
+// that should be compacted into a single new block.
+func (c *LeveledCompactor) selectOverlappingDirs(ds []dirMeta) []dirMeta {
+	if len(ds) < 2 {
+		return nil
+	}
+	startBlock, endBlock := -1, -1
```

Comment:
I think you need to use something else.

for example  -2 is a valid timestamp

Reply:
They are actually indices, where blocks from indices `startBlock` till `endBlock` (both inclusive) are overlapping.